internal/manager/conditions_ingressapi_test.go

```diff
@@ -0,0 +1,148 @@
+
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/manager"
+)
+
+type ingressTestOpts struct {
+	enabled      bool
+	crdInstalled bool
+}
+
+func TestIngressControllerConditions(t *testing.T) {
+	const kind = "Ingress"
+	var (
+		networkingV1 = schema.GroupVersionResource{
+			Group:    netv1.SchemeGroupVersion.Group,
+			Version:  netv1.SchemeGroupVersion.Version,
+			Resource: "ingresses",
+		}
+		networkingV1beta1 = schema.GroupVersionResource{
+			Group:    netv1beta1.SchemeGroupVersion.Group,
+			Version:  netv1beta1.SchemeGroupVersion.Version,
+			Resource: "ingresses",
+		}
+		extensionsV1beta1 = schema.GroupVersionResource{
+			Group:    extensionsv1beta1.SchemeGroupVersion.Group,
+			Version:  extensionsv1beta1.SchemeGroupVersion.Version,
+			Resource: "ingresses",
+		}
+	)
+
+	testCases := []struct {
+		name string
+
+		ingressNetV1      ingressTestOpts
+		ingressClassNetV1 ingressTestOpts
+		ingressNetV1beta  ingressTestOpts
+		ingressExtV1beta  ingressTestOpts
+
+		expectIngressNetV1      bool
+		expectIngressClassNetV1 bool
+		expectIngressNetV1beta  bool
+		expectIngressExtV1beta  bool
+		expectError             bool
+	}{
+		{
+			name:                    "netV1_takes_precedence_over_all",
+			ingressNetV1:            ingressTestOpts{true, true},
+			ingressClassNetV1:       ingressTestOpts{true, true},
+			ingressNetV1beta:        ingressTestOpts{true, true},
+			ingressExtV1beta:        ingressTestOpts{true, true},
+			expectIngressNetV1:      true,
+			expectIngressClassNetV1: true,
+		},
+		{
+			name:                   "netV1beta_wins_when_netV1_crds_not_installed",
+			ingressNetV1:           ingressTestOpts{true, false},
+			ingressClassNetV1:      ingressTestOpts{true, false},
+			ingressNetV1beta:       ingressTestOpts{true, true},
+			ingressExtV1beta:       ingressTestOpts{true, true},
+			expectIngressNetV1beta: true,
+		},
+		{
+			name:                   "extV1beta_wins_when_netV1_netV1beta_crds_not_installed",
+			ingressNetV1:           ingressTestOpts{true, false},
+			ingressClassNetV1:      ingressTestOpts{true, false},
+			ingressNetV1beta:       ingressTestOpts{true, false},
+			ingressExtV1beta:       ingressTestOpts{true, true},
+			expectIngressExtV1beta: true,
+		},
+		{
+			name:                   "netV1_not_picked_when_disabled",
+			ingressNetV1:           ingressTestOpts{false, true},
+			ingressClassNetV1:      ingressTestOpts{true, true},
+			ingressNetV1beta:       ingressTestOpts{true, true},
+			ingressExtV1beta:       ingressTestOpts{true, true},
+			expectIngressNetV1beta: true,
+		},
+		{
+			name:              "no_crds_installed",
+			ingressNetV1:      ingressTestOpts{true, false},
+			ingressClassNetV1: ingressTestOpts{true, false},
+			ingressNetV1beta:  ingressTestOpts{true, false},
+			ingressExtV1beta:  ingressTestOpts{true, false},
+			expectError:       true,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
```

Comment:
Ditto as elsewhere: do we save a lot by using `.Parallel()`? If not then let's get rid of it and get clean tests logs instead.

```suggestion
```

Reply:
[f4584b3](https://github.com/Kong/kubernetes-ingress-controller/pull/3013/commits/f4584b309593c92827e9c3adea19e9f020765591)